services/hierarchy/src/test/kotlin/ProjectServiceTest.kt

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright (C) 2024 The ORT Server Authors (See <https://github.com/eclipse-apoapsis/ort-server/blob/main/NOTICE>)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.eclipse.apoapsis.ortserver.services
+
+import io.kotest.core.spec.style.WordSpec
+import io.kotest.matchers.collections.shouldHaveSize
+import io.kotest.matchers.shouldBe
+
+import kotlinx.datetime.Clock
+
+import org.eclipse.apoapsis.ortserver.dao.test.DatabaseTestExtension
+import org.eclipse.apoapsis.ortserver.dao.test.Fixtures
+import org.eclipse.apoapsis.ortserver.model.OrtRun
+import org.eclipse.apoapsis.ortserver.model.RepositoryType
+import org.eclipse.apoapsis.ortserver.model.runs.AnalyzerConfiguration
+import org.eclipse.apoapsis.ortserver.model.runs.Environment
+import org.eclipse.apoapsis.ortserver.model.runs.Identifier
+import org.eclipse.apoapsis.ortserver.model.runs.ProcessedDeclaredLicense
+import org.eclipse.apoapsis.ortserver.model.runs.Project
+import org.eclipse.apoapsis.ortserver.model.runs.VcsInfo
+
+import org.jetbrains.exposed.sql.Database
+
+class ProjectServiceTest : WordSpec() {
+    private val dbExtension = extension(DatabaseTestExtension())
+
+    private lateinit var db: Database
+    private lateinit var fixtures: Fixtures
+
+    init {
+        beforeEach {
+            db = dbExtension.db
+            fixtures = dbExtension.fixtures
+        }
+
+        "listForOrtRunId" should {
+            "return the projects for the given ORT run ID" {
+                val service = ProjectService(db)
+
+                val projects = createProjects(
+                        listOf(
+                            Identifier("Maven", "com.example", "project1", "1.0.0"),
+                            Identifier("Maven", "com.example", "project2", "1.0.0")
+                        )
+                    )
+
+                val ortRunId = createAnalyzerRunWithProjects(projects).id
+                val result = service.listForOrtRunId(ortRunId).data
+
+                with(result.first()) {
+                    projects shouldHaveSize 2
+                    authors shouldHaveSize 2
```

Comment:
If you check for the content of the collection, you can omit the size check. You could also use matchers like `containsExactlyInAnyOrder`.